### PR TITLE
Refactor SRCaloPoint into its own file, add more info and add truth info

### PR DESCRIPTION
**Pull request by gputnam**

### CODE REVIEW

File: sbnanaobj/StandardRecord/SRCaloPoint.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      // inside a SRVector3D -- so just expose x/y/z out here
+      float x; //!< X-Position of deposition [cm]
+      float y; //!< Y-Position of deposition [cm]
+      float z; //!< Z-Position of deposition [cm]
+      float integral; //!< Hit Charge Integral [ADC]
+      float sumadc; //!< Hit Charge SummedADC [ADC]
+      float width; //!< Hit width [ticks]
+      short mult; //!< Hit multiplicity
+      short wire; //!< Wire of Calo-Point
+      short tpc; //!< TPC of Calo-Point
```
Comment (PetrilloAtWork) [inline sbnanaobj/StandardRecord/SRCaloPoint.h:34]: Does wire information make sense without the plane? and without the cryostat?

File: sbnanaobj/StandardRecord/SRCaloPoint.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace caf
+{
+
+  // Calo information on a point
+  class SRCaloPoint {
+    public:
+
+      SRCaloPoint();
+      virtual ~SRCaloPoint() {}
```
Comment (PetrilloAtWork) [inline sbnanaobj/StandardRecord/SRCaloPoint.h:17]: Destructor worse than useless. I know this is a pattern seen in other standard record classes... but nowhere it makes sense.
I recommend obliteration.

File: sbnanaobj/StandardRecord/SRTrueCaloPoint.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace caf
+{
+
+  // True calo information on a point
+  class SRTrueCaloPoint {
+
+    public:
+
+      SRTrueCaloPoint();
+      virtual ~SRTrueCaloPoint() {}
```
Comment (PetrilloAtWork) [inline sbnanaobj/StandardRecord/SRTrueCaloPoint.h:16]: Destructor obliteration recommended as above.

**Review by PetrilloAtWork — CHANGES_REQUESTED** [review]
A question about whether the "geolocation" of the point is complete enough to be useful...

Anyway, the only stopping issue from my side is that the new data members of `SRCaloPoint` (`width`, `mult`, `start`, `end`, `channel`) are not default-initialized. The rest is something that was inherited from the previous code.

File: sbnanaobj/StandardRecord/SRCaloPoint.cxx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  SRCaloPoint::SRCaloPoint():
+    rr(std::numeric_limits<float>::signaling_NaN()),
+    dqdx(std::numeric_limits<float>::signaling_NaN()),
+    dedx(std::numeric_limits<float>::signaling_NaN()),
+    pitch(std::numeric_limits<float>::signaling_NaN()),
+    t(std::numeric_limits<float>::signaling_NaN()),
+    integral(std::numeric_limits<float>::signaling_NaN()),
+    sumadc(std::numeric_limits<float>::signaling_NaN()),
+    wire(-1)
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbnanaobj/StandardRecord/SRCaloPoint.cxx]: Fixed.

File: sbnanaobj/StandardRecord/SRCaloPoint.h (hunk 2 above)
Comment (gputnam) [reply sbnanaobj/StandardRecord/SRCaloPoint.h]: Fixed.

File: sbnanaobj/StandardRecord/SRTrueCaloPoint.h (hunk 3 above)
Comment (gputnam) [reply sbnanaobj/StandardRecord/SRTrueCaloPoint.h]: Fixed.

**Review by PetrilloAtWork — APPROVED** [review]
If I recall correctly, the traditional uninitialised integral value was `-5` or `-999`, but I don't remember we ever agreed on that as a rule.
Moving to the other PR now...

**Pull request merged**